src/main/java/use_cases/attack_use_case/AttackInteractor.java

```diff
@@ -0,0 +1,121 @@
+package use_cases.attack_use_case;
+
+import entities.GameEvent;
+import entities.GameState;
+import entities.Player;
+import entities.cards.Card;
+import entities.cards.CreatureCard;
+import use_cases.GameOutputBoundary;
+import java.util.List;
+import java.util.ArrayList;
+
+public class AttackInteractor implements AttackInputBoundary {
+
+    private final GameState gameState;
+    private final AttackOutputBoundary attackPresenter;
+
+    public AttackInteractor(GameState gameState, AttackOutputBoundary attackPresenter) {
+        this.gameState = gameState;
+        this.attackPresenter = attackPresenter;
+    }
+
+    @Override
+    public void initiateAttack(AttackRequestModel inputData) {
+
+        Player player1 = gameState.getCurrentPlayer();
+        Player player2 = gameState.getOpposingPlayer();
+        CreatureCard creatureAttacker = player1.getCreatureById(inputData.getAttackerId());
+        if (player1.getNumOfEssence() < creatureAttacker.getAttackCost()) {
+            attackPresenter.displayFailMessage("You do not have enough essence.");
+        }
+        else {
+            List<Integer> possibleDefenders = new ArrayList<Integer>();
+            for (CreatureCard card : player2.getCreatures()) {
+                int cardId = card.getId();
+                if (cardId != inputData.getTargetId() & card.getDefendCost() <= player2.getNumOfEssence()) {
```

Comment:
It's better to use `&&` for this. A single `&` is bit-wise operator, basically if you write `A & B`, if A is false, it still checks B, but we know it's unnecessary because `A & B` will yield false anyway. If we use `A && B`, it will not bother checking B when it sees A is false already.
```suggestion
                if (cardId != inputData.getTargetId() && card.getDefendCost() <= player2.getNumOfEssence()) {
```